src/bank.rs

```diff
@@ -498,6 +516,11 @@ impl Bank {
             .iter_mut()
             .map(|a| (a.program_id, a.tokens))
             .collect();
+        let pre_userdata: Vec<_> = program_accounts
+            .iter_mut()
+            .map(|a| a.userdata.clone())
```

Comment:
I'm a little unsettled by this `userdata.clone()`, I feel like it'll hurt performance a little as this occurs on every transaction on every node even if are no account subscriptions.   But I have no data to back that up concretely.  Is there a quick way we can check if the node has at least one active account subscriptions, and if not skip all the cloning?    That's about the max I'd want to optimize here before actually measuring